module/zfs/zio.c

```diff
@@ -3130,15 +3131,15 @@ zio_write_gang_block(zio_t *pio, metaslab_class_t *mc)
 	boolean_t has_data = !(pio->io_flags & ZIO_FLAG_NODATA);
 
 	/*
-	 * If one copy was requested, store 2 copies of the GBH, so that we
-	 * can still traverse all the data (e.g. to free or scrub) even if a
-	 * block is damaged.  Note that we can't store 3 copies of the GBH in
-	 * all cases, e.g. with encryption, which uses DVA[2] for the IV+salt.
+	 * Store multiple copies of the GBH, so that we can still traverse
+	 * all the data (e.g. to free or scrub) even if a block is damaged.
+	 * This value respects the redundant_metadata property. Note that
+	 * we can't store 3 copies of the GBH in all cases, e.g. with
+	 * encryption, which uses DVA[2] for the IV+salt.
```

Comment:
The new code here does not enforce the encryption limitation, so the comment looks misplaces.